src/components/NcChip/NcChip.vue

```diff
@@ -162,22 +160,35 @@ const props = withDefaults(defineProps<{
 	variant: 'secondary',
 })
 
-const emit = defineEmits(['close'])
-const slots = useSlots()
+const emit = defineEmits<{
+	/**
+	 * Emitted when the close button is clicked
+	 */
+	close: []
+}>()
 
-const canClose = computed(() => !props.noClose)
-const hasActions = () => Boolean(slots.actions?.())
-const hasIcon = () => Boolean(props.iconPath || props.iconSvg || !!slots.icon?.())
+const slots = defineSlots<{
+	/**
+	 * The actions slot can be used to add custom actions (`NcAction*`) to the chips actions.
+	 */
+	actions: Slot
 
-/**
- * Handle closing the chip (pressing the X-button)
- */
-function onClose() {
 	/**
-	 * Emitted when the close button is clicked
+	 * The default slot can be used to set the text that is shown.
 	 */
-	emit('close')
-}
+	default: Slot
+
+	/**
+	 * The icon slot can be used to set the chip icon.
+	 * Make sure that the icon is not exceeding a height of `--chip-size`.
+	 * For round icons a exact size of `var(--chip-size)` is recommended.
+	 */
+	icon: Slot
+}>()
+
+const canClose = computed(() => !props.noClose)
+const hasActions = () => Boolean(slots.actions?.())
+const hasIcon = () => Boolean(props.iconPath || props.iconSvg || !!slots.icon?.())
```

Comment:
The same as in previous components, we should not render slots several times, and we moved from default to icon slots in such cases to make it easy for component users to pass the slot dynamically. It is only a problem fo the default slot.


```suggestion
const hasActions = () => !!slots.actions
const hasIcon = () => Boolean(props.iconPath || props.iconSvg || !!slots.icon)
```